vip-api/src/main/java/fr/insalyon/creatis/vip/api/security/ApiSecurityConfig.java

```diff
@@ -36,141 +36,105 @@
+ * - a static per-user API key (ApikeyAuthenticationFilter)
+ * - or an OIDC Bearer token (ex-Keycloak). This part is currently work-in-progress,
+ *   with org.keycloak currently removed, and proper OIDC connector not implemented yet.
  */
-@ComponentScan(basePackageClasses = {KeycloakSecurityComponents.class})
-@KeycloakConfiguration
+@Configuration
 @EnableWebSecurity
-@Order(1)
-public class ApiSecurityConfig extends KeycloakWebSecurityConfigurerAdapter {
+public class ApiSecurityConfig {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     private final Environment env;
-    private final ApikeyAuthenticationProvider apikeyAuthenticationProvider;
     private final VipAuthenticationEntryPoint vipAuthenticationEntryPoint;
-    private final VipKeycloakAuthenticationProvider vipKeycloakAuthenticationProvider;
+    private final AuthenticationManager vipAuthenticationManager;
 
     @Autowired
     public ApiSecurityConfig(
             Environment env, ApikeyAuthenticationProvider apikeyAuthenticationProvider,
-            VipAuthenticationEntryPoint vipAuthenticationEntryPoint,
-            VipKeycloakAuthenticationProvider vipKeycloakAuthenticationProvider) {
+            VipAuthenticationEntryPoint vipAuthenticationEntryPoint) {
         this.env = env;
-        this.apikeyAuthenticationProvider = apikeyAuthenticationProvider;
         this.vipAuthenticationEntryPoint = vipAuthenticationEntryPoint;
-        this.vipKeycloakAuthenticationProvider = vipKeycloakAuthenticationProvider;
-    }
-
-    protected boolean isKeycloakActive() {
-        return env.getProperty(KEYCLOAK_ACTIVATED, Boolean.class, Boolean.FALSE);
-    }
-
-    @Override
-    protected SessionAuthenticationStrategy sessionAuthenticationStrategy() {
-        // required for bearer only applications
-        return new NullAuthenticatedSessionStrategy();
-    }
-
-    @Override
-    protected AuthenticationEntryPoint authenticationEntryPoint() throws Exception {
-        return vipAuthenticationEntryPoint;
-    }
-
-    @Override
-    protected KeycloakAuthenticationProcessingFilter keycloakAuthenticationProcessingFilter() throws Exception {
-        KeycloakAuthenticationProcessingFilter f = super.keycloakAuthenticationProcessingFilter();
-        f.setAuthenticationFailureHandler(vipAuthenticationEntryPoint);
-        return f;
+        // Build our AuthenticationManager instance, with one provider for each authentication method
+        ArrayList<AuthenticationProvider> providers = new ArrayList<>();
+        providers.add(apikeyAuthenticationProvider);
+        // providers.add(oidcAuthenticationProvider);
+        this.vipAuthenticationManager = new ProviderManager(providers);
     }
 
-    @Override
-    protected KeycloakAuthenticationProvider keycloakAuthenticationProvider() {
-        return vipKeycloakAuthenticationProvider;
+    @Bean
+    public AuthenticationManager authenticationManager() {
+        return vipAuthenticationManager;
```

Comment:
It would be better to build the authenticationManager here instead of the constructor.

Reply:
Update, still needing some discussion : in https://github.com/ngeorges-cnrs/VIP-portal/commit/b398aae0fe387db14cbe9668e5d44679d5a82e49, I removed the authenticationManager Bean as I don't think it making this object a bean has any use. Wherever we decide to instantiate this object, the dependency chain with apikeyAuthenticationProvider and upcoming oidcAuthenticationProvider needs to be taken into account.